### ISSUE TITLE
Inconsistency with MIL loss function

### ISSUE THREAD

**Issue by bradezard131** [ppengtang/pcl.pytorch#28]
In the paper (and in the Caffe implementation I think), the MIL loss function is standard binary cross entropy, summed across classes, as in Bilen et. al.

In this implementation, you've done the mean across classes, rather than the sum. Could you explain this deviation from the original implementation?

**Comment by ppengtang** [comment]
Actually, in the Caffe implementation, I also use the mean across classes. I do this because I found that the sum will result in too large losses and worse mAPs.

**Comment by bradezard131** [comment]
It might be a good idea to note that the code does not match the paper then, I found the same but thought I must have had mistakes since your code could produce results with the details from the paper.

Regardless, I have managed to implement some version of PCL in my own repository now [here](https://github.com/bradezard131/wsod), using the latest Pytorch and Detectron2. I am still only achieving 44 mAP though, which is in line with the papers but not the reported result in this repository. Are there any other details here that are not in the PCL paper or the trick in the supplementary material of the region proposal paper? are you using any of the other things from the ECCV paper?

**Comment by ppengtang** [comment]
Hi, thanks for your implementation.

There is a small thing should be noticed:
unlike the Caffe implementation, in our PyTorch implementation, we use a different multi-scale training/testing strategy. More precisely, instead of resizing the shorter side of each image to a specified scale, we resize the longer side of each image to a specified scale, see [this line of code](https://github.com/ppengtang/pcl.pytorch/blob/d697e5f88b9232866a3a02554bba9fdfc6fa0d1a/lib/utils/blob.py#L132). We do this for two reasons: 1. This requires less GPU and is more efficient during training/testing; 2. We found this will give better results in the PyTorch implementation.

Other modifications compared with the Caffe codes and our papers are training iterations, batch size, warmup, and the two tricks referred to in #11 .

Please let me know if you get any updates and have any other questions.

**Comment by adityac8** [comment]
Hi @bradezard131 Thank you for such amazing work at https://github.com/bradezard131/wsod
I had some doubts while running the code but were unable to ask you as the issues tab is not activated for the repo.
Could you please open it so that I can post the doubts there.
Alternatively, could you please tell me your email id.
Thanks
